src/plots/smith/smith.js

```diff
@@ -0,0 +1,1456 @@
+        function() { return true; };
+
+    ax.setRange();
+    ax.setScale();
+    return ax;
+};
+
+proto.doAutoRange = function(fullLayout, smithLayout) {
+    var gd = this.gd;
+    var radialAxis = this.radialAxis;
+    var radialLayout = smithLayout.realaxis;
+
+    radialAxis.setScale();
+    doAutoRange(gd, radialAxis);
+
+    var rng = radialAxis.range;
+    radialLayout.range = rng.slice();
+    radialLayout._input.range = rng.slice();
+
+    radialAxis._rl = [
+        radialAxis.r2l(rng[0], null, 'gregorian'),
+        radialAxis.r2l(rng[1], null, 'gregorian')
+    ];
+};
+
+proto.updateRadialAxis = function(fullLayout, smithLayout) {
+    var _this = this;
+    var gd = _this.gd;
+    var layers = _this.layers;
+    var radius = _this.radius;
+    var cx = _this.cx;
+    var cy = _this.cy;
+    var radialLayout = smithLayout.realaxis;
+    var a0 = mod(smithLayout.sector[0], 360);
+    var ax = _this.radialAxis;
+    var hasRoomForIt = true;
+
+    _this.fillViewInitialKey('realaxis.angle', radialLayout.angle);
+    _this.fillViewInitialKey('realaxis.range', ax.range.slice());
+
+    ax.setGeometry();
+
+    // rotate auto tick labels by 180 if in quadrant II and III to make them
+    // readable from left-to-right
+    //
+    // TODO try moving deeper in Axes.drawLabels for better results?
+    if(ax.tickangle === 'auto' && (a0 > 90 && a0 <= 270)) {
+        ax.tickangle = 180;
+    }
+
+    var transFn2 = function(d) {
+        return strTranslate(ax.c2p(d.x), -2);
+    };
+
+    // set special grid path function
+    var gridPathFn = function(d) {
+        var value = d.x;
+
+        var gridRadius = 0.5 * (_this.radius - ax.c2p(value));
+        var gridCenter = gridRadius + ax.c2p(value);
+        return Lib.pathArc(gridRadius, 0, 2 * Math.PI, gridCenter, 0);
+    };
+
+    var newTickLayout = strTickLayout(radialLayout);
+    if(_this.radialTickLayout !== newTickLayout) {
+        layers['radial-axis'].selectAll('.xtick').remove();
+        _this.radialTickLayout = newTickLayout;
+    }
+
+    var labelFns = {};
+
+    labelFns.xFn = function() {
+        return 0;
+    };
+
+    labelFns.yFn = function() {
+        return 0;
+    };
+
+    labelFns.anchorFn = function() {
+        return 'end';
+    };
+
+    labelFns.heightFn = function() {
+        return 0;
+    };
+
+    if(hasRoomForIt) {
+        ax.setScale();
+
+        // circular grid lines
+        Axes.drawGrid(gd, ax, {
+            vals: [{x: 0.2 }, {x: 0.5 }, {x: 1.0 }, {x: 2.0 }, {x: 5.0 } ],
+            layer: layers['radial-grid'],
+            path: gridPathFn,
+            transFn: Lib.noop,
+            crisp: false
+        });
+
+        Axes.drawLabels(gd, ax, {
+            vals: [0.2, 0.5, 1.0, 2.0, 5.0].map(function(d) {
+                return {
+                    x: d,
+                    text: d,
+                    font: '"Open Sans", verdana, arial, sans-serif',
+                    fontColor: '#444',
+                    fontSize: 12,
```

Comment:
We should be able to use axis font size and color here.

Reply:
Not sure how to express this. `ax` here is a `mockAxis` declared on line 239 or so.

Reply:
Figured out how to thread font options through to the plot, in 33edd053530b139d36465b8c5d0463a41a986f07.